src/containers/Preview.js

```diff
@@ -27,12 +27,12 @@ function mapStateToProps(state) {
 
 function mapDispatchToProps(dispatch) {
   return {
-    onConsoleError(key, name, message) {
-      dispatch(consoleErrorProduced(key, name, message));
+    onConsoleError(key, name, message, projectKey) {
```

Comment:
Throughout this I would encourage you to use the language of `compiledProjectKey` rather than `projectKey`—they are keys for different entities!

Reply:
(Is there a better/less confusing name for the `CompiledProject` entity? Maybe `CompiledPage`?)